src/OrchardCore/OrchardCore.DisplayManagement/Entities/SectionDisplayDriver.cs

```diff
@@ -34,107 +38,111 @@ public override Task<IDisplayResult> DisplayAsync(TModel model, BuildDisplayCont
             return null;
         }
 
+        public override Task<IDisplayResult> EditAsync(TModel model, BuildEditorContext context)
+        {
+            var section = GetSection(model);
+
+            return EditAsync(model, section, context);
+        }
+
         public virtual Task<IDisplayResult> EditAsync(TModel model, TSection section, BuildEditorContext context)
         {
-            return EditAsync(section, context);
+            return Task.FromResult(Edit(model, section, context));
         }
 
-        public virtual Task<IDisplayResult> EditAsync(TSection section, BuildEditorContext context)
+        public virtual IDisplayResult Edit(TModel model, TSection section, BuildEditorContext context)
         {
-            return Task.FromResult(Edit(section, context));
+#pragma warning disable CS0618 // Type or member is obsolete
+            return Edit(section, context);
+#pragma warning restore CS0618 // Type or member is obsolete
         }
 
+        [Obsolete("This method will be removed. Instead use EditAsync(TModel model, TSection section, BuildEditorContext context) or Edit(TModel model, TSection section, BuildEditorContext context)")]
         public virtual IDisplayResult Edit(TSection section, BuildEditorContext context)
         {
             return Edit(section);
         }
 
+        [Obsolete("This method will be removed. Instead use EditAsync(TModel model, TSection section, BuildEditorContext context) or Edit(TModel model, TSection section, BuildEditorContext context)")]
         public virtual IDisplayResult Edit(TSection section)
-        {
-            return null;
-        }
+            => NullShapeResult();
 
-        public virtual Task<IDisplayResult> UpdateAsync(TModel model, TSection section, IUpdateModel updater, UpdateEditorContext context)
+        public override async Task<IDisplayResult> UpdateAsync(TModel model, UpdateEditorContext context)
         {
-            return UpdateAsync(section, updater, context);
+            var section = GetSection(model);
+
+            var result = await UpdateAsync(model, section, context);
+
+            if (result == null)
+            {
+                return null;
+            }
+
+            if (context.Updater.ModelState.IsValid)
+            {
+                model.Properties[SectionName] = JObject.FromObject(section);
+            }
+
+            return result;
         }
 
-        public virtual Task<IDisplayResult> UpdateAsync(TSection section, IUpdateModel updater, UpdateEditorContext context)
+        public virtual Task<IDisplayResult> UpdateAsync(TModel model, TSection section, UpdateEditorContext context)
         {
+#pragma warning disable CS0618 // Type or member is obsolete
             return UpdateAsync(section, context);
+#pragma warning restore CS0618 // Type or member is obsolete
         }
 
+        [Obsolete("This method will be removed. Instead use UpdateAsync(TModel model, TSection section, UpdateEditorContext context)")]
```

Comment:
Since there's no such method, I believe you meant `UpdateAsync(TModel model, UpdateEditorContext context)`. Also for the other `Obsolete` below.

Reply:
The referenced method exists on line 113.